subiquity/ui/views/filesystem/filesystem.py

```diff
@@ -378,106 +378,44 @@ def refresh_model_inputs(self):
         log.debug('FileSystemView: building device list')
         rows = []
 
-        def _usage_label(obj):
-            cd = obj.constructed_device()
-            if cd is not None:
-                return _("{component_name} of {name}").format(
-                    component_name=cd.component_name, name=cd.name)
-            fs = obj.fs()
-            if fs is not None:
-                if not self.parent.model.is_mounted_filesystem(fs.fstype):
-                    return _("formatted as {fstype}").format(
-                        fstype=fs.fstype)
-                else:
-                    m = fs.mount()
-                    if m:
-                        return _(
-                            "formatted as {fstype}, mounted at {path}").format(
-                                fstype=fs.fstype, path=m.path)
-                    else:
-                        return _("formatted as {fstype}, not mounted").format(
-                            fstype=fs.fstype)
-            else:
-                return _("unused")
-
-        rows.append(TableRow([Color.info_minor(heading) for heading in [
-            Text(" "),
-            Text(_("DEVICE")),
-            Text(_("SIZE"), align="center"),
+        rows.append(Color.info_minor(TableRow([
+            Text(""),
+            (2, Text(_("DEVICE"))),
             Text(_("TYPE")),
-            Text(" "),
-            Text(" "),
-        ]]))
+            Text(_("SIZE"), align="center"),
+            Text(""),
+            Text(""),
+        ])))
         for device in devices:
-            menu = self._action_menu_for_device(device)
-            label = device.label
-            if device.annotations:
-                label = "{} ({})".format(label, ", ".join(device.annotations))
-            cells = [
-                Text("["),
-                Text(label),
-                Text("{:>9}".format(humanize_size(device.size))),
-                Text(device.desc()),
-                menu,
-                Text("]"),
-            ]
-            row = make_action_menu_row(cells, menu)
-            rows.append(row)
-
-            if not device.partitions():
+            for obj, cells in summarize_device(
+                    device,
+                    lambda part: part.available() == self.show_available):
+                if obj is not None:
+                    menu = self._action_menu_for_device(obj)
+                else:
+                    menu = Text("")
+                if obj is device:
+                    start, end = '[', ']'
+                else:
+                    start, end = '', ''
+                cells = [Text(start)] + cells + [menu, Text(end)]
+                if obj is not None:
+                    rows.append(make_action_menu_row(cells, menu))
+                else:
+                    rows.append(TableRow(cells))
```

Comment:
Style wise `if obj is not None:` is redundand. One can simply use `if obj:`. Also, I normally preffer avoiding to use the else clause, and instead initialize things.... ie.

```
menu = Text("")
start, end = '', ''
if obj:
    menu = self.....
if obj is device:
    start, end = '[', ']'
```

But that's just style knitpicks. Functionality looks fine.